### PR TITLE
Ensure null pointer is popped if unused

### LINKED ISSUE #7550: ? : with Byte* causes Invalid Program with Optimize+
I was glancing at another bug report (regarding ?. with a Byte*) that I can't find now and I adjusted the code (in LinqPad) to get it to compile and it threw InvalidProgramException.

I tried with Optimize- and it works fine. Looking at the IL, the optimized version clearly "forgets" to pop the unused value on the stack.

    //Straight from C# Program in LinqPad:
    void Main()
    {
        unsafe
        {
            var x = new A();
            byte* ptr = (x == null) ? null : x.Ptr;
        }
    }
    
    // Define other methods and classes here
    public class A
    {
        public unsafe byte* Ptr;
    }

The optimised IL displayed by the same program:

    IL_0000:  newobj      UserQuery+A..ctor
    IL_0005:  stloc.0     // x
    IL_0006:  ldloc.0     // x
    IL_0007:  brfalse.s   IL_0011
    IL_0009:  ldloc.0     // x
    IL_000A:  ldfld       UserQuery+A.Ptr
    IL_000F:  pop         
    IL_0010:  ret         
    IL_0011:  ldc.i4.0    
    IL_0012:  conv.u      
    IL_0013:  ret         
    
    A..ctor:
    IL_0000:  ldarg.0     
    IL_0001:  call        System.Object..ctor
    IL_0006:  ret         

IMHO There should be a pop after IL_0012.

This is with LinqPad v5.03.07 and as I mentioned at http://stackoverflow.com/q/32665389/256431 I don't have a simple way to confirm the C# compiler version used, but I do know it is Rosyln-based.

### PR DESCRIPTION
**Pull request by leppie**
Fixes #7550

### CODE REVIEW

**Comment by leppie** [comment]
retest prtest/lin/dbg/unit32 please

**Comment by davkean** [comment]
tag @dotnet/roslyn-compiler 

test eta please

File: src/Compilers/CSharp/Test/Emit/CodeGen/UnsafeTests.cs [hunk 1]
```diff
@@ -9194,6 +9194,29 @@ static void Main(string[] args)
             var result = CompileAndVerify(compilation, expectedOutput: "5");
         }
 
+
+        [Fact]
```
Comment (cston) [inline src/Compilers/CSharp/Test/Emit/CodeGen/UnsafeTests.cs]: Please add `[WorkItem(7550)]`.

**Comment by cston** [comment]
LGTM

File: src/Compilers/CSharp/Test/Emit/CodeGen/UnsafeTests.cs (hunk 1 above)
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Emit/CodeGen/UnsafeTests.cs]: ```[WorkItem(7550, "https://github.com/dotnet/roslyn/issues/7550")]```

**Comment by AlekseyTs** [comment]
LGTM, modulo the lack of WorkItem attribute.

**Comment by leppie** [comment]
@AlekseyTs Updated commit with WorkItem attribute.

**Comment by TyOverby** [comment]
LGTM

**Comment by VSadov** [comment]
LGTM

**Comment by AlekseyTs** [comment]
LGTM

**Pull request merged**